cypress/e2e/project-list.cy.ts

```diff
@@ -59,3 +59,34 @@ describe("Project List", () => {
     });
   });
 });
+
+describe("Error Handling", () => {
+  it(
+    "Displays error message on failed fetch request",
+    { defaultCommandTimeout: 12000 },
+    () => {
+      cy.intercept("GET", "https://prolog-api.profy.dev/project", {
+        forceNetworkError: true,
+      }).as("failedFetch");
+
+      cy.visit("http://localhost:3000/dashboard", {
+        retryOnNetworkFailure: false,
+      });
+
+      cy.wait("@failedFetch");
+
+      cy.get('[data-cy="alert-error"]', { timeout: 12000 }).should(
+        "be.visible",
+      );
+      cy.get('[data-cy="alert-error"]').find("button").click();
+      cy.intercept("GET", "https://prolog-api.profy.dev/project").as(
+        "fetchRequest",
+      );
+      cy.wait("@fetchRequest").then((interception) => {
+        expect(interception.request.url).to.equal(
+          "https://prolog-api.profy.dev/project",
+        );
+      });
+    },
+  );
```

Comment:
Great job with this test. Just fyi below is the test I wrote which doesn't require intercepting the succeeding request (note the `times: 4`)

```
it("shows an error message", () => {
    // setup request mock
    cy.intercept(
      { url: "https://prolog-api.profy.dev/project", times: 4 },
      {
        statusCode: 500,
      },
    );

    // open issues page
    cy.visit("http://localhost:3000/dashboard/issues");

    // check that the error message is shown
    cy.get("[data-testid='issues-error-message']", { timeout: 15000 })
      .should("be.visible")
      // click retry button
      .find("button")
      .click();

    cy.get("[data-testid='issue-list']").should("be.visible");
  });
```